src/legacy/core_plugins/kibana/public/visualize/saved_visualizations/_saved_vis.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * @name SavedVis
+ *
+ * @extends SavedObject.
+ *
+ * NOTE: It's a type of SavedObject, but specific to visualizations.
+ */
+// @ts-ignore
+import { Vis } from 'ui/vis';
+import { SavedObject, SavedObjectKibanaServices } from 'ui/saved_objects/types';
+import { createSavedObjectClass } from 'ui/saved_objects/saved_object';
+import { updateOldState } from '../../../../visualizations/public';
+import { VisualizeConstants } from '../visualize_constants';
+import { extractReferences, injectReferences } from './saved_visualization_references';
+import { IndexPattern } from '../../../../../../plugins/data/public';
```

Comment:
You can use `IIndexPattern` here and in `src/legacy/ui/public/saved_objects/types.ts` (tested this locally)

Reply:
thx! done!